router/keeper/keeper.go

```diff
@@ -33,6 +33,33 @@ type Keeper struct {
 	distrKeeper    types.DistributionKeeper
 }
 
+type PacketMetadata struct {
+	Forward *ForwardMetadata `json:"forward"`
+}
+
+type ForwardMetadata struct {
+	Receiver string          `json:"receiver"`
+	Port     string          `json:"port"`
+	Channel  string          `json:"channel"`
+	Timeout  time.Duration   `json:"timeout"`
+	Retries  *uint8          `json:"retries"`
+	Next     *PacketMetadata `json:"next"`
```

Comment:
I may be wrong, but AFAICT this will prevent us from generating metadata where `next` is something other than a PacketMetadata (i.e.: `"forward": ...`). Can we have this as `[]byte` and deserialize it in two steps? that way if the forward key is not present we can still send that packet and allow the receiver to process its metadata

Reply:
yes good call, it should probably be consistent with the memo field, so I changed it to a string.